packages/onboarding/src/setup-tailored-site-after-creation.ts

```diff
@@ -93,13 +93,37 @@ export function setupSiteAfterCreation( { siteId, flowName }: SetupOnboardingSit
 		return Promise.resolve();
 	};
 
+	const removeInitialPost = async ( siteId: number, flowName: string ) => {
+		if ( flowName === NEWSLETTER_FLOW ) {
+			// We are deleting the initial "Hello World!" post for newsletter sites
+			// The id for the first post on a new site will be 3
+
+			// First request will move post to trash
+			await wpcomRequest( {
+				path: `/sites/${ siteId }/posts/3/delete?http_envelope=1`,
+				method: 'POST',
+				apiNamespace: 'rest/v1.1',
+			} );
+
+			// Second request will permanently delete the post
+			await wpcomRequest( {
+				path: `/sites/${ siteId }/posts/3/delete?http_envelope=1`,
+				method: 'POST',
+				apiNamespace: 'rest/v1.1',
+			} );
+		}
+
+		return Promise.resolve();
+	};
+
 	if ( siteId && flowName ) {
 		return Promise.all( [
 			postSiteSettings( siteId ),
 			postSiteLogo(),
 			setPattern( siteId, flowName ),
 			setIntent( siteId, flowName ),
 			setLaunchpadScreen( siteId, flowName ),
+			removeInitialPost( siteId, flowName ),
```

Comment:
Might want to repeat here for less verbose repetition.
```suggestion
			// First request will move post to trash
			removeInitialPost( siteId, flowName ).then(() => {
				 // Second request will permanently delete the post 
				 return removeInitialPost( siteId, flowName )
			})
```